src/app/shared/model/audio-version.model.ts

```diff
@@ -3,19 +3,18 @@ import { HalDoc, Upload } from '../../core';
 import { BaseModel } from './base.model';
 import { AudioFileModel } from './audio-file.model';
 import { VERSION_TEMPLATED } from './invalid';
+import { HasUpload, applyMixins } from './upload';
 
-export class AudioVersionModel extends BaseModel {
+export class AudioVersionModel extends BaseModel implements HasUpload {
 
   static DEFAULT_LABEL = 'Main Audio';
 
   public id: number;
-  public uploads: string = '';
   public label: string;
   public explicit: string = '';
-  public files: AudioFileModel[];
 
   // save in-progress uploads to localstorage
-  SETABLE = ['uploads', 'label', 'explicit'];
+  SETABLE = ['label', 'explicit', 'hasUploadMap'];
```

Comment:
It's kind of a bummer that I have to put `hasUploadMap` here.  And re-define the other 3 methods from that mixin below.  But unfortunately that's the only way to make typescript play nice with them.